ostelco-ios-client/Coordinators/OnboardingCoordinator.swift

```diff
@@ -218,6 +215,52 @@ class OnboardingCoordinator {
         }
         return false
     }
+    
+    /**
+     Tries to download and install an eSIM on the device using apples eSIM APIs.
+     */
+    func addPlanForSimProfile(_ simProfile: SimProfile) -> PromiseKit.Promise<Void> {
+        let planObj = CTCellularPlanProvisioning()
```

Comment:
We should probably make a thing to handle this rather than add the functionality directly to the coordinator, no? The act of adding an esim I mean.

Reply:
Where do you suggest we put this logic?

Reply:
I mean contain it into it's own struct/class where we can give it a sim and ask it to do the work. The onboarding coordinator's job is to handle changing steps in the on boarding flow, not to install esims.

Reply:
I've moved the addPlan to a separate file.